### PR TITLE
Add join-semilattice

### PR DESCRIPTION
**Pull request by Hakimba**
Hi, going hand in hand with the meet-semilattice, this PR add the join-semilattice.

### CODE REVIEW

File: .github/dot/specs.dot [hunk 1]
```diff
@@ -73,4 +73,5 @@ graph {
         Semigroup -- Arrow_alt [style=dotted; arrowhead=none];
 
         Meet_semilattice -- Semigroup [style=dotted; arrowhead=none; constraint=false];
+        Join_semilattice -- Semigroup [style=dotted; arrowhead=none; constraint=false];
```
Comment (xvw) [inline .github/dot/specs.dot:76]: I suggest to add the same relation betweenn join and meet

**Review by xvw — APPROVED** [review]
Very nice. Thanks !

File: .github/dot/specs.dot (hunk 1 above)
Comment (Hakimba) [reply .github/dot/specs.dot:76]: done, tell me if its ok now

**Review by xvw — COMMENTED** [review]
Yes, i approve. 
Please @gr-im make a review!

**Review by xvw — APPROVED** [review]

**Review by gr-im — DISMISSED** [review]

**Review by gr-im — APPROVED** [review]

**Pull request merged**